test/session_tests/DocumentQueryTest.ts

```diff
@@ -8,220 +8,493 @@ import {RequestExecutor} from "../../src/Http/Request/RequestExecutor";
+        })
+        .whereIn('name', 'withNesting').get();
+
+      expect(results).to.have.lengthOf(10);
+
+      results.map(function (obj) {
+        expect(obj.name).to.equal('withNesting');
+      });
 
-      expect(results[0].order).to.equals('d');
     });
 
-    it('should paginate', async() => {
-      const expectedUids: number[][] = [[2,3],[4,5],[6,90],[95]];
-      const pageSize: number = 2;
-      
-      const totalCount: number = await store.openSession().query<Product>({
-        documentType: Product,
-        waitForNonStaleResults: true
-      }).whereNotNull('uid').count();
+    it('should query with whereGreaterThan', async () => {
+      const results: Universal[] = await store.openSession()
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true
+        })
+        .whereGreaterThan('order', 2).get();
 
-      const totalPages: number = Math.ceil(totalCount / pageSize);
+      expect(results).to.have.lengthOf(8);
 
-      expect(totalCount).to.equals(7);
-      expect(totalPages).to.equals(4);
+      results.map(function (obj) {
+        expect(obj.order).to.not.be.lessThan(2);
+        expect(obj.order).to.be.greaterThan(2);
+      });
 
-      for (let page: number = 1; page <= totalPages; page++) {
-        const products: Product[] = await store.openSession().query<Product>({
-          documentType: Product,
-          waitForNonStaleResults: true
-        }).whereNotNull('uid').orderBy('uid').skip((page - 1) * pageSize).take(pageSize).get();
+    });
+
+    it('should query with whereLessThan', async () => {
+      const results: Universal[] = await store.openSession()
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true
+        })
+        .whereLessThan<number>('order', 6).get();
+
+      expect(results).to.have.lengthOf(5);
+
+      results.map(function (obj) {
+        expect(obj.order).to.not.be.greaterThan(6);
+        expect(obj.order).to.be.lessThan(6);
+      });
 
-        expect(products).to.have.length.lte(pageSize);
-        products.forEach((product: Product, index: number) => expect(product.uid).to.equals(expectedUids[page - 1][index]));
-      }
     });
 
-    it('should query with includes', async() => {
-      session = store.openSession();
+    it('should query with whereLessThan with :p', async () => {
+      const results: Universal[] = await store.openSession()
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true,
+          queryParameters: {
+            "p0" : 6
+          }
+        })
+        .whereLessThan<string>('order', ':p0').get();
+
+      expect(results).to.have.lengthOf(5);
 
-      await session.query({
-        waitForNonStaleResults: true, 
-        includes: ['product_id']
-      }).where({uid: 92}).get();
+      results.map(function (obj) {
+        expect(obj.order).to.not.be.greaterThan(6);
```

Comment:
That's what I was talking about when commenting `whereGreaterOrEquals` 👍 Please add this kind of assertions where needed.